server/src/main/java/org/opensearch/index/mapper/IpFieldMapper.java

```diff
@@ -222,25 +224,51 @@ protected Object parseSourceValue(Object value) {
 
         @Override
         public Query termQuery(Object value, @Nullable QueryShardContext context) {
-            failIfNotIndexed();
+            failIfNotIndexedAndNoDocValues();
+            Query query;
             if (value instanceof InetAddress) {
-                return InetAddressPoint.newExactQuery(name(), (InetAddress) value);
+                query = InetAddressPoint.newExactQuery(name(), (InetAddress) value);
             } else {
                 if (value instanceof BytesRef) {
                     value = ((BytesRef) value).utf8ToString();
                 }
                 String term = value.toString();
                 if (term.contains("/")) {
                     final Tuple<InetAddress, Integer> cidr = InetAddresses.parseCidr(term);
-                    return InetAddressPoint.newPrefixQuery(name(), cidr.v1(), cidr.v2());
+                    query = InetAddressPoint.newPrefixQuery(name(), cidr.v1(), cidr.v2());
+                } else {
+                    InetAddress address = InetAddresses.forString(term);
+                    query = InetAddressPoint.newExactQuery(name(), address);
                 }
-                InetAddress address = InetAddresses.forString(term);
-                return InetAddressPoint.newExactQuery(name(), address);
             }
+            if (isSearchable() && hasDocValues()) {
+                return new IndexOrDocValuesQuery(
+                    query,
+                    SortedSetDocValuesField.newSlowRangeQuery(
```

Comment:
@harshavamsi this is a single term query, shouldn't we use `SortedSetDocValuesField.newSlowExactQuery` instead?

Reply:
ah you're right